test/integration/master/audit_dynamic_test.go

```diff
@@ -0,0 +1,247 @@
+		require.NoError(t, err, "Failed to match all expected events, events %#v not found!", missing)
+	})
+
+	t.Run("two sink", func(t *testing.T) {
+		// reset event list
+		eventList1.Store(auditinternal.EventList{})
+
+		_, err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Create(&sink2)
+		require.NoError(t, err, "failed to create audit sink2")
+		t.Log("created audit sink2")
+
+		// sleep to wait for sink to provision
+		time.Sleep(5 * time.Second)
+
+		// perform configmap ops
+		configMapOperations(t, kubeclient)
+
+		// check for corresponding events in both sinks
+		missing, err := checkForEvents(t, eventList1)
+		require.NoError(t, err, "Failed to match all expected events for sink1, events %#v not found!", missing)
+
+		missing, err = checkForEvents(t, eventList2)
+		require.NoError(t, err, "Failed to match all expected events for sink2, events %#v not found!", missing)
+	})
+
+	t.Run("delete sink", func(t *testing.T) {
+		err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Delete("test2", &metav1.DeleteOptions{})
+		require.NoError(t, err, "failed to delete audit sink2")
+		t.Log("deleted audit sink2")
+
+		// sleep to wait for sink to close
+		time.Sleep(5 * time.Second)
+
+		// reset event lists
+		eventList1.Store(auditinternal.EventList{})
+		eventList2.Store(auditinternal.EventList{})
+
+		// perform configmap ops
+		configMapOperations(t, kubeclient)
+
+		// check for corresponding events
+		missing, err := checkForEvents(t, eventList1)
+		require.NoError(t, err, "Failed to match all expected events for sink1, events %#v not found!", missing)
+		require.Len(t, eventList2.Load().(auditinternal.EventList).Items, 0, "eventList2 should be empty")
+	})
+
+	t.Run("update sink", func(t *testing.T) {
+		sink1, err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Get("test1", metav1.GetOptions{})
+		require.NoError(t, err)
+		sink1.Spec.Webhook.ClientConfig.URL = &server2.URL
+		_, err = kubeclient.AuditregistrationV1alpha1().AuditSinks().Update(sink1)
+		require.NoError(t, err, "failed to update audit sink1")
+		t.Log("updated audit sink1")
+
+		// sleep to wait for sink to close
+		time.Sleep(5 * time.Second)
+
+		// reset event lists
+		eventList1.Store(auditinternal.EventList{})
+		eventList2.Store(auditinternal.EventList{})
+
+		// perform configmap ops
+		configMapOperations(t, kubeclient)
+
+		// check for corresponding events
+		missing, err := checkForEvents(t, eventList2)
+		require.NoError(t, err, "Failed to match all expected events for sink2, events %#v not found!", missing)
+		require.Len(t, eventList1.Load().(auditinternal.EventList).Items, 0, "eventList1 should be empty")
+	})
+}
+
+// checkForEvents will poll to check for the expected audit events in an atomic event list
+func checkForEvents(t *testing.T, a *atomic.Value) ([]utils.AuditEvent, error) {
+	var missing []utils.AuditEvent
+	err := wait.Poll(1*time.Second, 120*time.Second, func() (bool, error) {
```

Comment:
use PollImmediate with a much shorter interval (50ms?)... I'd expect the events to show up on the order of milliseconds